src/components/modals/LogsModal.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react'
+import { Platform, View, ViewStyle } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import RNFS from 'react-native-fs'
+import Share, { ShareOptions } from 'react-native-share'
+
+import { MultiLogOutput } from '../../actions/LogActions'
+import s from '../../locales/strings'
+import { sendLogs } from '../../modules/Logs/api'
+import { WarningCard } from '../cards/WarningCard'
+import { CrossFade } from '../common/CrossFade'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow, ModalMessage, ModalTitle } from '../themed/ModalParts'
+import { OutlinedTextInput } from '../themed/OutlinedTextInput'
+import { ThemedModal } from '../themed/ThemedModal'
+interface Props {
+  bridge: AirshipBridge<void>
+  logs: MultiLogOutput
+}
+
+const SENSITIVE_KEY_REGEX = /(allKeys|displayPrivateSeed|displayPublicSeed|otpKey|loginKey|recoveryKey|dataKey|syncKey)/g
+
+export const LogsModal = (props: Props) => {
+  const { bridge, logs } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const [userMessage, setUserMessage] = React.useState('')
+
+  const isDangerous = React.useMemo(() => {
+    return SENSITIVE_KEY_REGEX.test(logs.activity.data) || SENSITIVE_KEY_REGEX.test(logs.info.data)
+  }, [logs])
+
+  const handleShare = async () => {
+    const payload = JSON.stringify(logs, null, 2)
+    const username = logs.info.loggedInUser?.userName ?? ''
+    const dir = Platform.OS === 'android' ? RNFS.ExternalCachesDirectoryPath : RNFS.DocumentDirectoryPath
+    const path = `${dir}/edge-log-${username}.json`.replace('-.json', '.json')
+
+    const shareOptions: ShareOptions = {
+      title: s.strings.settings_button_export_logs,
+      subject: s.strings.settings_button_export_logs,
+      message: s.strings.settings_button_export_logs,
+      urls: [`file://${path}`],
+      type: 'application/json',
+      failOnCancel: false
+    }
+
+    await RNFS.writeFile(path, payload, 'utf8').catch((e: any) => {
+      showError(e.message)
+    })
+    return await Share.open(shareOptions)
+      .catch((e: any) => {
+        showError(e.message)
+      })
+      .then(() => {
+        bridge.resolve()
+      })
+  }
+
+  const handleSend = async () => {
+    logs.info.userMessage = userMessage
+    logs.activity.userMessage = userMessage
+
+    return await Promise.all([
+      sendLogs(logs.activity).catch((e: any) => {
+        showError(`${s.strings.settings_modal_send_logs_failure} activity logs code ${e.message}`)
+      }),
+      sendLogs(logs.info).catch((e: any) => {
+        showError(`${s.strings.settings_modal_send_logs_failure} info logs code ${e.message}`)
+      })
```

Comment:
These `.catch` statements can stay, since they are actually providing a translated message. But do `e?.message` in both of them, for extra safety. We don't want to crash if somebody does `throw null`.